Add Jazzy distribution as pre-release

marcoag opened the pull request:
As part of the work towards [branching Jazzy from Rolling](https://discourse.ros.org/t/preparing-for-jazzy-branching-from-rolling-2024-04-18/37260) using the [rosdistro migration tools](https://github.com/ros/rosdistro/tree/master/migration-tools) we need an empty target distribution for `Jazzy`.

The first commit of this PR updates de index files and adds the `distribution.yaml` file for `jazzy` which will get re-populated when the migration tool runs.

I'd like to get review of this pull request in two phases.

I'd like on review phase to approve the platform list and source ref as I prepare to run the migration tool. This PR won't be merged as-is once approved. Instead it will be used as the foundation for a migration using the following migration command:
```
python3 migration-tools/migrate-rosdistro.py --source rolling --dest jazzy \
  --source-ref ad3bf8c1330cea014e30e4c68c2e318dfbd43b6e \
  --release-org ros2-gbp \
```
Once the migration is complete, I'll update this PR and request a second phase of reviews for merging the completed migration.

marcoag committed:
Initial yaml and index updates for Jazzy.

Signed-off-by: Marco A. Gutierrez <marcogg@marcogg.com>

clalancette commented on index-v4.yaml:
```diff
@@ -81,6 +81,12 @@ distributions:
+  jazzy:
+    distribution: [jazzy/distribution.yaml]
+    distribution_cache: http://repo.ros2.org/rosdistro_cache/jazzy-cache.yaml.gz
+    distribution_status: active
```
We should set this to pre-release for now:

```suggestion
    distribution_status: pre-release
```

(see also #36923 , where we did this for Iron)

nuclearsandwich commented on jazzy/distribution.yaml:
```diff
@@ -0,0 +1,14 @@
+  - '9'
+  ubuntu:
+  - noble
+repositories: []
```
```suggestion
repositories: {}
```
Repositories is associative / dictish / mapish not an array.

marcoag committed:
Update index-v4.yaml

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

marcoag committed:
Update jazzy/distribution.yaml

Co-authored-by: Steven! Ragnarök <nuclearsandwich@users.noreply.github.com>

nuclearsandwich commented:
CI wants an empty list of repositories to be an empty value. But the migration script will only work if there's an empty map there instead.
So we can leave CI angry about this for now.

```
 --- file:///home/runner/work/rosdistro/rosdistro/jazzy/distribution.yaml
+++ loaded-and-saved
@@ -11,3 +11,3 @@
   - noble
-repositories: []
+repositories:
 type: distribution
```

clalancette reviewed (commented):
This looks good to me now.  I'm not going to approve it, and just leave that for @nuclearsandwich to do as needed.

nuclearsandwich reviewed (approved):
https://butt.holdings and also https://www.youtube.com/watch?v=861jOEqVr4Q

Approved to begin migration!

marcoag committed:
Jazzy distribution.yaml after first bloom attempt.

Signed-off-by: Marco A. Gutierrez <marcogg@marcogg.com>

clalancette committed:
Remove robotnt_msgs.

The 'ros2-rolling-devel' branch was removed, and there
is no obvious replacement branch.  Remove it for now.

Signed-off-by: Chris Lalancette <clalancette@gmail.com>
